athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/metadata/ListTablesRequest.java

```diff
@@ -32,23 +32,50 @@
         extends MetadataRequest
 {
     private final String schemaName;
+    private final String nextToken;
+    private final int pageSize;
 
     /**
      * Constructs a new ListTablesRequest object.
      *
      * @param identity The identity of the caller.
      * @param queryId The ID of the query requesting metadata.
      * @param catalogName The catalog name that tables should be listed for.
-     * @param schemaName The schema name that tables should be listed for. This may be null if no specific schema is requested.
+     * @param schemaName The schema name that tables should be listed for. This may be null if no specific schema is
+     *                   requested.
      */
     @JsonCreator
     public ListTablesRequest(@JsonProperty("identity") FederatedIdentity identity,
-            @JsonProperty("queryId") String queryId,
-            @JsonProperty("catalogName") String catalogName,
-            @JsonProperty("schemaName") String schemaName)
+                             @JsonProperty("queryId") String queryId,
+                             @JsonProperty("catalogName") String catalogName,
+                             @JsonProperty("schemaName") String schemaName)
+    {
+        this(identity, queryId, catalogName, schemaName, null, -1);
+    }
+
+    /**
+     * Constructs a new ListTablesRequest object.
+     *
+     * @param identity The identity of the caller.
+     * @param queryId The ID of the query requesting metadata.
+     * @param catalogName The catalog name that tables should be listed for.
+     * @param schemaName The schema name that tables should be listed for. This may be null if no specific schema is
+     *                   requested.
+     * @param nextToken The pagination starting point for the next page (null indicates the first paginated request).
+     * @param pageSize The page size used for pagination (-1 indicates the request should not be paginated).
+     */
+    @JsonCreator
+    public ListTablesRequest(@JsonProperty("identity") FederatedIdentity identity,
+                             @JsonProperty("queryId") String queryId,
+                             @JsonProperty("catalogName") String catalogName,
+                             @JsonProperty("schemaName") String schemaName,
+                             @JsonProperty("nextToken") String nextToken,
+                             @JsonProperty("pageSize") int pageSize)
```

Comment:
why do you need two constructors? you should have only one, this is the tricky part of forwards and backwards compatibility

Reply:
before you submit for a another review, please test forwards and backwards compatibility (e.g. old client calling new lambda, new client calling old lambda, and new client calling new lambda)

Reply:
We are still using `version 2` of the SDK. Removing the original constructor will be the opposite of backwards compatibility. It would mean that many many files will have to change, not just in the SDK, but for every single connector as well as the control plane.

Reply:
Removed original constructor and tested all scenarios using a local stack.

Reply:
Can you attach testing evidence to the PR (e.g. what test scenarios you ran and how you confirmed they were successful).

Reply:
evidence provided offline